set.mm

```diff
@@ -88969,14 +89034,74 @@ being the empty set ( ` A e/ _V ` ).  (Contributed by AV, 6-Aug-2019.) $)
   diffi $p |- ( A e. Fin -> ( A \ B ) e. Fin ) $=
     ( cfn wcel cdif wss difss ssfi mpan2 ) ACDABEZAFJCDABGAJHI $.
 
-  ${
-    $d A x $.  $d X x $.  $d M x $.
+  $( Lemma for ~ rexdif1en and ~ dif1en .  (Contributed by BTernaryTau,
+     18-Aug-2024.) $)
+  dif1enlem $p |- ( ( F e. V /\ M e. _om /\ F : A -1-1-onto-> suc M ) ->
+      ( A \ { ( `' F ` M ) } ) ~~ M ) $=
+    ( wcel com wf1o cfv csn cdif cres wa wfo adantr wb mpbird c0 adantl syl3anc
+    wceq csuc w3a ccnv cen wbr simp1 sucidg wfun dff1o3 simprbi f1ofo wfn f1ofn
+    fnresdm foeq1 wf wne f1ocnvdm f1ocnvfv2 snidg eqeltrd fressnfv biimp3ar cin
+    3syl disjsn con2bii sylib fnresdisj syl mtbid neqned foconst syl2anc resdif
+    wn sylan2 word nnord orddif f1oeq3d ancoms 3adant1 cvv resexg f1oen3g sylan
+    ) BDEZCFEZACUAZBGZUBWHACBUCZHZIZJZCBWOKZGZWOCUDUEZWHWIWKUFWIWKWQWHWKWIWQWKW
+    ILWQWOWJCIZJZWPGZWIWKCWJEZXACFUGWKXBLZWLUHZAWJBAKZMZWNWSBWNKZMZXAWKXDXBWKAW
+    JBMZXDAWJBUIUJNWKXFXBWKXFXIAWJBUKWKBAULZXEBTXFXIOAWJBUMZABUNAWJXEBUOVEPNXCW
+    NWSXGUPZXGQUQXHXCXJWMAEZWMBHZWSEZXLWKXJXBXKNAWJCBURZXCXNCWSAWJCBUSXBCWSEWKC
+    WJUTRVAXJXMXLXOAWMWSBVBVCSXCXGQXCAWNVDQTZXGQTZXCXMXQVPXPXQXMAWMVFVGVHWKXQXR
+    OZXBWKXJXSXKAWNBVIVJNVKVLWNCXGVMVNAWNWJWSBVOSVQWIWQXAOWKWICWTWOWPWICVRCWTTC
+    VSCVTVJWARPWBWCWHWPWDEWQWRBWODWEWOCWPWDWFWGVN $.
+  $( $j usage 'dif1enlem' avoids 'ax-pow'; $)
+
+  ${
+    $d A f x $.  $d M f x $.
+    $( If a set is equinumerous to a nonzero finite ordinal, then there exists
+       an element in that set such that removing it leaves the set equinumerous
+       to the predecessor of that ordinal.  (Contributed by BTernaryTau,
+       26-Aug-2024.) $)
+    rexdif1en $p |- ( ( M e. _om /\ A ~~ suc M ) ->
+        E. x e. A ( A \ { x } ) ~~ M ) $=
+      ( vf csuc cen wbr com wcel cv wf1o wex csn cdif wrex bren 19.42v ccnv cvv
+      wa cfv sucidg f1ocnvdm ancoms sylan vex dif1enlem mp3an1 wceq sneq breq1d
+      difeq2d rspcev syl2anc exlimiv sylbir sylan2b ) BCEZFGCHIZBURDJZKZDLZBAJZ
+      MZNZCFGZABOZBURDPUSVBTUSVATZDLVGUSVADQVHVGDVHCUTRUAZBIZBVIMZNZCFGZVGUSCUR
+      IZVAVJCHUBVAVNVJBURCUTUCUDUEUTSIUSVAVMDUFBUTCSUGUHVFVMAVIBVCVIUIZVEVLCFVO
+      VDVKBVCVIUJULUKUMUNUOUPUQ $.
+    $( $j usage 'rexdif1en' avoids 'ax-pow'; $)
+  $}
+
+  ${
+    $d A f $.  $d M f $.  $d X f $.
     $( If a set ` A ` is equinumerous to the successor of a natural number
        ` M ` , then ` A ` with an element removed is equinumerous to ` M ` .
        (Contributed by Jeff Madsen, 2-Sep-2009.)  (Revised by Stefan O'Rear,
-       16-Aug-2015.) $)
+       16-Aug-2015.)  Avoid ~ ax-pow .  (Revised by BTernaryTau,
+       26-Aug-2024.) $)
     dif1en $p |- ( ( M e. _om /\ A ~~ suc M /\ X e. A ) ->
         ( A \ { X } ) ~~ M ) $=
```

Comment:
Why is this, as well as the current version, that long to prove ? And why do we need M finite ?

It looks simpler than that, unless I'm missing something:

Suppose $M \in On$ and $A \simeq suc M$. Then, there is a bijection $f \colon suc M = M \cup \\{M\\} \to A$. Then, prove that $f \setminus \\{ \langle \\{M\\} , \\{f(M)\\} \rangle \\}$ is a bijection from $M$ to $A \setminus \\{f(M)\\}$. Finally, compose with the transposition of $f(M)$ and $X$ in $A$ (see my above comment).

Reply:
M doesn't need to be finite, I just kept it that way to match the original theorem. The main complication with your approach is that the composition must be proven to exist without using ax-pow (like coexg does). I actually prioritized working on findcard and findcard2 so that it would be easier to prove the finite case for theorems like coexg without using ax-pow, but in the meantime I got around this by using restriction and union instead of composition.

Reply:
Since the original proof is along the same non-optimal lines, this thread is **non-blocking**.

As you noticed, general existence of the composition of two functions requires ax-pow in our formalization, but in specific cases, one can avoid ax-pow by explicitly **constructing** the function we want, as you did in the statement of the theorem above, that is, 

$f|_{M \setminus f^{-1}(X)} \cup \left( x \in \\{f^{-1}(X)\\} \mapsto f(M) \right) \colon M \to A \setminus \\{f(M)\\}$

is a bijection, without finiteness assumption.

Reply:
Disregarding the change from $M \in \omega$ to $M \in On$, I think changing f1ofvswap to mostly match your approach means that our overall approaches are roughly the same? Is there a separate change that you were suggesting be made to dif1en or dif1enlem that I'm just missing?

Reply:
I haven't looked at your proof in non-compressed format, so I'm not sure, but above I wrote "as you did in the statement...", so I guess the function used is the same. And the main point of this thread is the finiteness assumption anyway.

I'm a bit surprised by the length of the proof, but if you're confident you proceeded as in the first comment of this thread, then it's good for me (though I don't see in that proof where finiteness would even help). Maybe that's a matter of available "library lemmas".